File: liqp-core/src/main/java/liqp/nodes/KeyValueNode.java
package liqp.nodes;

import com.google.common.collect.ImmutableList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import liqp.node.LNode;
import lombok.Getter;

@Getter
public class KeyValueNode implements LNode {

    public final String key;
    public final LNode value;

    public KeyValueNode(String key, LNode value) {
        this.key = key;
        this.value = value;
    }

    @Override
    public Object render(LContext context) {
        Map<String, Object> map = new HashMap<String, Object>();
        map.put(key, value.render(context));
        return map;
    }

  @Override
  public List<LNode> children() {
    return ImmutableList.of(value);
  }
}

File: liqp-core/src/main/kotlin/liqp/nodes/KeyValueNode.kt
package liqp.nodes

import liqp.context.LContext
import liqp.node.LNode

class KeyValueNode(val key: String, val value: LNode) : LNode() {

  override val children = listOf(value)

  override fun render(context: LContext): Any {
    return mutableMapOf<Any, Any?>().apply {
      this[key] = value.render(context)
    }
  }
}
